### DISCUSSION TITLE
Custom integration ist installiert. Was nun?

### DISCUSSION THREAD

**Discussion by therealspider67** [gickowtf/pixoo-homeassistant#3] [Q&A] [answered]
Moin!
Ich habe jetzt den neuen Unterordner mit der Integration. Dein Beispiel habe ich in die Config.Yaml am Ende eingefügt und die IP Adresse auf die von meinem Pixoo64 geändert. Einen der Werte für die Solaranzeige habe ich ebenfalls angepasst um zu sehen ob der Sensor angezeigt wird. Das ganze geprüft und gespeichert und anschließend neu gestartet. 
Es passiert bei mir leider nichts auf dem Pixoo. Was mache ich falsch?
Was noch interessant wäre: Wie bekomme ich meine eigenen Pixoo Bilder ins System? Ich kann sie ja auf dem Handy oder Tablet malen und dann? Bei mir passt das mit den Solar Icons nicht, da ich keinen Speicher habe. Ich habe nur Aktuelle Einspeisung, Aktueller Verbrauch, Zählerstand, Zurück ins Netz.

**Comment by gickowtf** [comment, marked as answer]
Moin, probiere bitte zu erst eine ganz simple config:

```
divoom_pixoo:
  ip_address: 'CHANGEME'
  scan_interval:
    seconds: 15
  pages:
    - page: 1
      texts:
        - text: "TEST"
          position: [5, 10] # 5 nach rechts und 10 nach unten
          font: FONT_GICKO
          font_color: [255, 0, 0]  # red
```

Wenn das funktioniert funktioniert auch alles andere.

Bei der PV anzeige ist es zwingend erforderlich das alle Keywords korrekt gefüllt sind. 

```
power: "{{ states('sensor.deinSensor') | float(0) }}"
storage: "{{ states('sensor.deinSensor') | float(0) }}"
discharge: "{{ states('sensor.deinSensor') | float(0) }}"
powerhousetotal: "{{ states('sensor.deinSensor') | float(0) }}"
vomNetz: "{{ states('sensor.deinSensor') | float(0) }}"
time: "{{ now().strftime('%H:%M') }}"  #Format HH:MM der kann so bleiben wie er hier steht der zeigt die Aktuelle Uhrzeit
```

viel Erfolg ;)

**Reply by therealspider67** [reply]
Der Test hat funktioniert. Ich habe jetzt anstelle des Textes meinen Currend Sensor eingefügt, und auch der wird angezeigt.Anschließend habe ich es mit den Sensoren Power:, Storage: usw. versucht und in alle Felder  den gleichen sensor von mir eingegeben. Hier wurde mir leider nichts angezeigt. Ich schaffe es nur einen Wert auf dem Pixoo anzuzeigen. sowie ich einen zusätzlichen Wert oder Text anzeigen möchte streikt der Pixoo.
Am idealsten wäre Eine von dir erstellte Eingabemaske wo mann zwei Spalten hat. die erste Spalte sollte man frei benennen können(natürlich nur am vorhandenen Platz gemessen) in der zweiten Spalte trägt man seinen Sensor ein. So hat man eine Bezeichnung und einen Sensor. Dann müsste man noch weitere Zeilen anlegen können die genauso funktionieren. Die einzelnen Spalten sollte man noch in der Farbe ändertn können. Wernn man jetzt noch ein Bild in den Hintertgrund legen könnte wäre das schon sehr viel Anwenderfreundlicher.
Wäre sowas in der Art machbar?

**Reply by therealspider67** [reply]
divoom_pixoo:
  ip_address: '192.168.178.81'
  scan_interval:
    seconds: 15
  pages:
    - page: 1
      texts:
        - text: "{{ states('sensor.tasmota_sml_power_curr') | float(0) }}"
          position: [34, 10] # 34 nach rechts und 10 nach unten
          font: FONT_GICKO
          font_color: [0, 0, 255]  # blau   
        - text: "Akt.Verbrauch"
          position: [0, 4]
          font: FONT_PICO_8
          font_color: [0, 255, 0]  # grün
        - text: "Solareinspeisung"
          position: [0, 20]
          font: FONT_PICO_8
          font_color: [0, 255, 0]  # grün 
        - text: "{{ states('sensor.balkonsolar_power') | float(0) }}"
          position: [34, 26] # 34 nach rechts und 26 nach unten
          font: FONT_GICKO
          font_color: [0, 0, 255]  # blau
        - text: "Zaehlerstand"
          position: [0, 36]
          font: FONT_PICO_8
          font_color: [0, 255, 0]  # grün 
        - text: "{{ states('sensor.tasmota_sml_total_in') | float(0) }}"
          position: [34, 42] # 34 nach rechts und 42 nach unten
          font: FONT_GICKO
          font_color: [0, 0, 255]  # blau  
 #     images:
  #      - image: "/homeassistant/custom_components/divoom_pixoo/img/sunpower.png" #max 64 x 64 Pixel
  #        position: [0, 0]   
          
    - page: 2
      PV:
        - power: "{{ states.sensor.balkonsolar_power.state }}"
          storage: "{{ states.sensor.tasmota_sml_power_curr.state }}"
          discharge: "{{ states.sensor.tasmota_sml_power_curr.state }}"
          powerhousetotal: "{{ states.sensor.tasmota_sml_power_curr.state }}"
          vomNetz: "{{ states.sensor.tasmota_sml_total_in.state }}"
          time: "{{ now().strftime('%H:%M') }}"  #Format HH:MM
Das hat jetzt hingehauen, allerdings kann ich kein Bild einfügen. Siehe oben ( Habe Rauten davor gemacht.)

**Comment by therealspider67** [comment]
![IMG_20231119_112212](https://github.com/gickowtf/pixoo-homeassistant/assets/141012342/6f87bdcc-6878-4f8f-8b7b-2454baa369d5)
![IMG_20231119_112222](https://github.com/gickowtf/pixoo-homeassistant/assets/141012342/11d81589-cb77-441f-bbcd-79c7ca202d1e)

**Reply by gickowtf** [reply]
so wie es aussieht funktioniert alles?

**Reply by therealspider67** [reply]
Hallo, danke der Nachfrage. Ich Versuche krampfhaft ein Pictogramm aus deiner Bibliothek auf meinen Pixoo zu zaubern. Bei PV funktioniert das (siehe Bild) allerdings habe ich nur ein Balkonkraftwerk und keinen Speicher. Deshalb habe ich dort einfach nochmal den aktuellen Verbrauch eingesetzt damit das Ganze läuft. Ich würde gerne vor meine Werte Pictogramme einfügen. und die Beschreibungen dann weglassen. Habe das wie in dem Fettgedruckten Oben versucht, bekomme aber das Bild nicht ausgegeben.
`images:
- image: "/homeassistant/custom_components/divoom_pixoo/img/sunpower.png" #max 64 x 64 Pixel
position: [0, 0]`

Ich habe mich an deinem Beispiel hier bei GitHub orientiert. Mit dem dort angegebenen Pfad funktioniert das bei mir auch nicht.
Bei dir steht:
`  images:
        - image: "/config/img/anyPicture.png" #max 64 x 64 Pixel
          position: [30, 10]`
Den Pfad gibt es bei mir meines Wissens nicht.
![IMG_20231121_143918](https://github.com/gickowtf/pixoo-homeassistant/assets/141012342/8b508f64-4e19-4232-9faa-8e5ffcf596f4)
So sieht das bei mir jetzt aus. Mit den Pictogrammen würde es ähnlich wie in deinem Beispiel aussehen

**Reply by gickowtf** [reply]
```
    - page: 2
      texts:
        - text: "github/gickowtf"
          position: [0, 10]
          font: FONT_PICO_8
          font_color: [255, 0, 0]
        - text: "Thx 4 Support"
          position: [0, 30]
          font: FONT_PICO_8
          font_color: [255, 0, 0]
      images:
        - image: "/config/custom_components/divoom_pixoo/img/sunpower.png"
          position: [30, 10]
```

Teste es mal mit diesem Pfad und Bild

**Reply by therealspider67** [reply]
![IMG_20231122_085444](https://github.com/gickowtf/pixoo-homeassistant/assets/141012342/14a32ac6-ba95-411c-ba32-e601930de296)
Es hat geklappt !!! Vielen herzlichen Dank, damit kann ich jetzt alles realisieren was ich möchte. 
LG Spidy

**Reply by gickowtf** [reply]
Yeah nice! Hast du die Bilder selbst gemacht? Würde mich über einen Stern freuen :)

**Reply by therealspider67** [reply]
Ja, habe ich selbst gemacht, bin aber noch nicht damit zufrieden. (Pixel Art Studio Free)
Hast du schonmal überlegt einen Lauftext im unteren oder oberen Bildbereich zu programmieren?

**Reply by therealspider67** [reply]
![IMG_20231123_083443](https://github.com/gickowtf/pixoo-homeassistant/assets/141012342/7aa1e45f-fef7-4227-9525-99bec648e8fb)
So finde ich das etwas stylischer
